tests/integration/integration_test.go

```diff
@@ -1514,6 +1515,31 @@ func TestDynamicPythonVenv(t *testing.T) {
 	})
 }
 
+// Tests dynamic provider in Python using input/output types.
+func TestDynamicPythonTypes(t *testing.T) {
+	var randomVal string
+	var randomSize float64
+	integration.ProgramTest(t, &integration.ProgramTestOptions{
+		Dir: filepath.Join("dynamic", "python_types"),
+		Dependencies: []string{
+			filepath.Join("..", "..", "sdk", "python", "env", "src"),
+		},
+		ExtraRuntimeValidation: func(t *testing.T, stack integration.RuntimeValidationStackInfo) {
+			randomVal = stack.Outputs["random_val"].(string)
+			randomSize = stack.Outputs["random_size"].(float64)
+		},
+		EditDirs: []integration.EditDir{{
+			Dir:      "step1",
+			Additive: true,
+			ExtraRuntimeValidation: func(t *testing.T, stack integration.RuntimeValidationStackInfo) {
+				assert.Equal(t, randomVal, stack.Outputs["random_val"].(string))
+				assert.Equal(t, randomSize, stack.Outputs["random_size"].(float64))
+			},
+		}},
+		UseAutomaticVirtualEnv: true,
+	})
+}
+
 func TestResourceWithSecretSerialization(t *testing.T) {
```

Comment:
Is it valuable to have a full integration test that is structurally similar to what real SDKs will do as well?

Reply:
Yes, definitely. I plan to add more tests, including integration tests. It's one of the remaining tasks.